Build failure: linuxPackages.v4l2loopback

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Use a master after latest staging-next merge https://github.com/NixOS/nixpkgs/pull/432489

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/305810283

### Relevant log output

```console
Running phase: unpackPhase
unpacking source archive /nix/store/45rsam0bsa3ap5ax7jxy7w5066ki2bgi-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
substituteStream() in derivation v4l2loopback-0.15.1-6.12.43: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'Makefile')
build flags: SHELL=/nix/store/4bacfs7zrg714ffffbjp57nsvcz6zfkq-bash-5.3p3/bin/bash CC=/nix/store/qdknxw57cwy1jkrhq7fzmiis73j42jv6-gcc-14.3.0/bin/gcc LD=/nix/store/xrwdb41dqi2ia6lr2s61w5bzfg2m71pi-binutils-2.44/bin/ld AR=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/ar NM=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/nm STRIP=/nix/store/xrwdb41dqi2ia6lr2s61w5bzfg2m71pi-binutils-2.44/bin/strip OBJCOPY=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/objcopy OBJDUMP=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/objdump READELF=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/readelf HOSTCC=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/cc HOSTCXX=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/c++ HOSTAR=/nix/store/ky2x48xfid0nn5arablpwlnsngj70gws-binutils-wrapper-2.44/bin/ar HOSTLD=/nix/store/ky2x48xfid0nn5arablpwlnsngj70gws-binutils-wrapper-2.44/bin/ld ARCH=x86_64 CROSS_COMPILE= KBUILD_OUTPUT=/nix/store/ljkg7hla8yvf1q5vvqp0abmspm7s6x58-linux-6.12.43-dev/lib/modules/6.12.43/build KERNELRELEASE=6.12.43 KERNEL_DIR=/nix/store/ljkg7hla8yvf1q5vvqp0abmspm7s6x58-linux-6.12.43-dev/lib/modules/6.12.43/build
grep: /etc/dkms/framework.conf: No such file or directory
grep: /etc/dkms/framework.conf.d/*.conf: No such file or directory

++++++ To sign the  module, you must set KBUILD_SIGN_KEY/KBUILD_SIGN_CERT to point to the signing key/certificate!
++++++ For your convenience, we try to read these variables as 'mok_signing_key' resp. 'mok_certificate' from /etc/dkms/framework.conf /etc/dkms/framework.conf.d/*.conf

++++++ If your certificate requires a password, pass it via the KBUILD_SIGN_PIN env-var!
grep: /etc/dkms/framework.conf: No such file or directory
grep: /etc/dkms/framework.conf.d/*.conf: No such file or directory
++++++ E.g. using 'export KBUILD_SIGN_PIN; read -s -p "Passphrase for signing key : " KBUILD_SIGN_PIN; sudo --preserve-env=KBUILD_SIGN_PIN make sign'

Building v4l2-loopback driver...
make -C /nix/store/ljkg7hla8yvf1q5vvqp0abmspm7s6x58-linux-6.12.43-dev/lib/modules/6.12.43/build M=/build/source KCPPFLAGS="" modules
make[1]: Entering directory '/nix/store/ljkg7hla8yvf1q5vvqp0abmspm7s6x58-linux-6.12.43-dev/lib/modules/6.12.43/build'
  CC [M]  /build/source/v4l2loopback.o
  MODPOST /build/source/Module.symvers
  CC [M]  /build/source/v4l2loopback.mod.o
  CC [M]  /build/source/.module-common.o
  LD [M]  /build/source/v4l2loopback.ko
  BTF [M] /build/source/v4l2loopback.ko
Skipping BTF generation for /build/source/v4l2loopback.ko due to unavailability of vmlinux
make[1]: Leaving directory '/nix/store/ljkg7hla8yvf1q5vvqp0abmspm7s6x58-linux-6.12.43-dev/lib/modules/6.12.43/build'
make -C utils V4L2LOOPBACK_SNAPSHOT_VERSION=
make[1]: Entering directory '/build/source/utils'
/nix/store/qdknxw57cwy1jkrhq7fzmiis73j42jv6-gcc-14.3.0/bin/gcc  -I..  -c -o v4l2loopback-ctl.o v4l2loopback-ctl.c
/nix/store/qdknxw57cwy1jkrhq7fzmiis73j42jv6-gcc-14.3.0/bin/gcc   v4l2loopback-ctl.o   -o v4l2loopback-ctl
/nix/store/xrwdb41dqi2ia6lr2s61w5bzfg2m71pi-binutils-2.44/bin/ld: cannot find crt1.o: No such file or directory
/nix/store/xrwdb41dqi2ia6lr2s61w5bzfg2m71pi-binutils-2.44/bin/ld: cannot find crti.o: No such file or directory
/nix/store/xrwdb41dqi2ia6lr2s61w5bzfg2m71pi-binutils-2.44/bin/ld: cannot find -lgcc_s: No such file or directory
collect2: error: ld returned 1 exit status
make[1]: *** [<builtin>: v4l2loopback-ctl] Error 1
make[1]: Leaving directory '/build/source/utils'
make: *** [Makefile:97: utils/v4l2loopback-ctl] Error 2
```

### Additional context

The build fails since staging-next; channel for nixos-unstable is not yet advanced, but as staging-next is already merged, so i still wanted to report the build issue

FYI: In staging-next includes a change which kernel build flags are passed, see https://github.com/NixOS/nixpkgs/pull/402198 . 

I assume that this related as zfs module had an issue and the error looked quite similar.
The code fix for zfs does not apply here, as zfs is using autoconf. For zfs he mapping for those flags needed so be changed.

v4l2loopback is more simple, but possibly its stiil and issue with the switched compiler (wrapped -> unwrapped)?


If i see it correctly we fail on building the user space utils and the kernelmodule builds fine.



### System metadata

❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.41, NixOS, 25.11 (Xantusia), 25.11.20250819.cf34cee`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - channels(shawn): `""`
 - nixpkgs: `/nix/store/a5g6b7pkp4dc9czd73bsx7n8w56bh5d1-source`


### Notify maintainers

@moni-dz 
@Bot-wxt1221 


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/v4l2loopback/default.nix
+++ b/pkgs/os-specific/linux/v4l2loopback/default.nix
@@ -32,6 +32,11 @@ stdenv.mkDerivation {
     sed -i '/depmod/d' Makefile
   '';
 
+  # Don't use makeFlags for this
+  postBuild = ''
+    make utils
+  '';
+
   nativeBuildInputs = [ kmod ] ++ kernel.moduleBuildDependencies;
 
   postInstall = ''
@@ -46,6 +51,7 @@ stdenv.mkDerivation {
   makeFlags = kernelModuleMakeFlags ++ [
     "KERNELRELEASE=${kernel.modDirVersion}"
     "KERNEL_DIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "v4l2loopback.ko"
   ];
 
   meta = {

